prog/rand.go

```diff
@@ -26,6 +26,10 @@ func (r *randGen) rand(n int) uintptr {
 	return uintptr(r.Intn(n))
 }
 
+func (r *randGen) randRange(begin int, end int) uintptr {
+	return uintptr(begin + r.Intn(end-begin))
```

Comment:
I would expect that array 2:3 means [2:3] boundaries inclusive, rather than fixed-size array of size 2. This code makes it a fixed-size array. Check randGen.randRangeInt function.

Speaking of this, please also update sys/README.md to describe range-sized arrays and note there if the range is inclusive, exclusive of half-open.